tmk_core/common/mousekey.c

```diff
@@ -95,10 +100,82 @@ static uint8_t wheel_unit(void) {
     return (unit > MOUSEKEY_WHEEL_MAX ? MOUSEKEY_WHEEL_MAX : (unit == 0 ? 1 : unit));
 }
 
+#else /* #ifndef MK_KINETIC_SPEED */
+
+/*
+ * Kinetic movement  acceleration algorithm
+ *
+ *  current speed = I + A * T/50 + A * 0.5 * T^2 | maximum B
+ *
+ * T: time since the mouse movement started
+ * E: mouse events per second (set through MOUSEKEY_INTERVAL, UHK sends 250, the
+ *    pro micro on my Signum 3.0 sends only 125!)
+ * I: initial speed at time 0
+ * A: acceleration
+ * B: base mouse travel speed
+ */
+const uint16_t mk_accelerated_speed = MOUSEKEY_ACCELERATED_SPEED;
+const uint16_t mk_base_speed = MOUSEKEY_BASE_SPEED;
+const uint16_t mk_decelerated_speed = MOUSEKEY_DECELERATED_SPEED;
+const uint16_t mk_initial_speed = MOUSEKEY_INITIAL_SPEED;
+
+static uint8_t move_unit(void) {
+    float speed = mk_initial_speed;
+
+    if (mousekey_accel & ((1<<0) | (1<<2))) {
+        speed = mousekey_accel & (1<<2) ? mk_accelerated_speed : mk_decelerated_speed;
+    } else if (mousekey_repeat && mouse_timer) {
+        const float time_elapsed = timer_elapsed(mouse_timer) / 50;
+        speed = mk_initial_speed +
+            MOUSEKEY_MOVE_DELTA * time_elapsed +
+            MOUSEKEY_MOVE_DELTA * 0.5 * time_elapsed * time_elapsed;
+
+        speed = speed > mk_base_speed ? mk_base_speed : speed;
+    }
+
+    /* convert speed to USB mouse speed 1 to 127 */
+    speed = (uint8_t)(speed / (1000.0f / mk_interval));
+    speed = speed < 1 ? 1 : speed;
+
+    return speed > MOUSEKEY_MOVE_MAX ? MOUSEKEY_MOVE_MAX : speed;
+}
+
+float mk_wheel_interval = 1000.0f / MOUSEKEY_WHEEL_INITIAL_MOVEMENTS;
```

Comment:
```suggestion
const float mk_wheel_interval = 1000.0f / MOUSEKEY_WHEEL_INITIAL_MOVEMENTS;
```

Add `const` in this line.

Reply:
`mk_wheel_interval` is mutated here and therefore can't be made constant: https://github.com/qmk/qmk_firmware/blob/4d25747bf4638d825a72dd1f91940e234f731739/tmk_core/common/mousekey.c#L163

Reply:
It's true, well observed